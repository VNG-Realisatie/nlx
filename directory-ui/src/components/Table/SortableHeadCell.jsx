import React from 'react'
import { oneOf } from 'prop-types'
import { StyledSortableTableHeadCell, StyledArrow } from './SortableHeadCell.styles'

export const ASCENDING = 'asc'
export const DESCENDING = 'desc'

const ArrowUp = () =>
  <StyledArrow viewBox="0 0 8 10">
    <path d="M5 6V0H3v6H0l4 4 4-4z" fill="#2D3240" />
  </StyledArrow>

const ArrowDown = () =>
  <StyledArrow viewBox="0 0 8 10">
    <path d="M3 4v6h2V4h3L4 0 0 4z" fill="#2D3240" />
  </StyledArrow>

const SortableArrow = ({ direction }) =>
  direction === ASCENDING ? <ArrowDown/> : <ArrowUp/>

const SortableHeadCell = ({ children, direction, ...other }) =>
  <StyledSortableTableHeadCell isSorting={direction} {...other}>
    {children}
    {direction ? <SortableArrow direction={direction} /> : null}
  </StyledSortableTableHeadCell>

SortableHeadCell.propTypes = {
  direction: oneOf([ASCENDING, DESCENDING, null])
}

export default SortableHeadCell
