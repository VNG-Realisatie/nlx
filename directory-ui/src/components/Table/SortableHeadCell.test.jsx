import React from 'react'
import { shallow, mount } from 'enzyme'
import SortableHeadCell from './SortableHeadCell'

const addToTable = headCell =>
  <table><thead><tr>{headCell}</tr></thead></table>

it('should match snapshot', () => {
  expect(shallow(<SortableHeadCell/>)).toMatchSnapshot()
})

it('should support the ascending direction', () => {
  const wrapper = addToTable(<SortableHeadCell direction="asc"/>)
  expect(mount(wrapper)).toMatchSnapshot()
})

it('should support the descending direction', () => {
  const wrapper = addToTable(<SortableHeadCell direction="desc"/>)
  expect(mount(wrapper)).toMatchSnapshot()
})
