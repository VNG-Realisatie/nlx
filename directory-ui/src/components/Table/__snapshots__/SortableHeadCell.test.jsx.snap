// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Styled(styled.th)
  align="left"
/>
`;

exports[`should support the ascending direction 1`] = `
<table>
  <thead>
    <tr>
      <SortableHeadCell
        direction="asc"
      >
        <Styled(styled.th)
          align="left"
        >
          <StyledComponent
            align="left"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "faglPq",
                  "rules": Array [
                    "
    display: table-cell;
    padding: .65rem;

    font-size: .85rem;
    line-height: 1.25rem;
    font-weight: 600;
    color: #ADB3C6;
    overflow: hidden;
    text-overflow: ellipsis;
    text-transform: uppercase;
    white-space: nowrap;
    
    text-align: ",
                    [Function],
                    ";
",
                    "
    cursor: pointer;
",
                  ],
                },
                "defaultProps": Object {
                  "align": "left",
                },
                "displayName": "Styled(styled.th)",
                "foldedComponentIds": Array [
                  "sc-bdVaJa",
                ],
                "propTypes": Object {
                  "align": [Function],
                  "children": [Function],
                },
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "th",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <th
              className="sc-bdVaJa sc-bwzfXH faglPq"
            >
              <SortableArrow
                direction="asc"
              >
                <ArrowDown>
                  <styled.svg
                    viewBox="0 0 8 10"
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-htpNat",
                            "isStatic": true,
                            "lastClassName": "fxfxjB",
                            "rules": Array [
                              "
  width: 8px;
  height: 10px;
  position: relative;
  left: 5px;
  top: 1px;
",
                            ],
                          },
                          "displayName": "styled.svg",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-htpNat",
                          "target": "svg",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      viewBox="0 0 8 10"
                    >
                      <svg
                        className="sc-htpNat fxfxjB"
                        viewBox="0 0 8 10"
                      >
                        <path
                          d="M3 4v6h2V4h3L4 0 0 4z"
                          fill="#2D3240"
                        />
                      </svg>
                    </StyledComponent>
                  </styled.svg>
                </ArrowDown>
              </SortableArrow>
            </th>
          </StyledComponent>
        </Styled(styled.th)>
      </SortableHeadCell>
    </tr>
  </thead>
</table>
`;

exports[`should support the descending direction 1`] = `
<table>
  <thead>
    <tr>
      <SortableHeadCell
        direction="desc"
      >
        <Styled(styled.th)
          align="left"
        >
          <StyledComponent
            align="left"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "faglPq",
                  "rules": Array [
                    "
    display: table-cell;
    padding: .65rem;

    font-size: .85rem;
    line-height: 1.25rem;
    font-weight: 600;
    color: #ADB3C6;
    overflow: hidden;
    text-overflow: ellipsis;
    text-transform: uppercase;
    white-space: nowrap;
    
    text-align: ",
                    [Function],
                    ";
",
                    "
    cursor: pointer;
",
                  ],
                },
                "defaultProps": Object {
                  "align": "left",
                },
                "displayName": "Styled(styled.th)",
                "foldedComponentIds": Array [
                  "sc-bdVaJa",
                ],
                "propTypes": Object {
                  "align": [Function],
                  "children": [Function],
                },
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "th",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <th
              className="sc-bdVaJa sc-bwzfXH faglPq"
            >
              <SortableArrow
                direction="desc"
              >
                <ArrowUp>
                  <styled.svg
                    viewBox="0 0 8 10"
                  >
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-htpNat",
                            "isStatic": true,
                            "lastClassName": "fxfxjB",
                            "rules": Array [
                              "
  width: 8px;
  height: 10px;
  position: relative;
  left: 5px;
  top: 1px;
",
                            ],
                          },
                          "displayName": "styled.svg",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-htpNat",
                          "target": "svg",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      viewBox="0 0 8 10"
                    >
                      <svg
                        className="sc-htpNat fxfxjB"
                        viewBox="0 0 8 10"
                      >
                        <path
                          d="M5 6V0H3v6H0l4 4 4-4z"
                          fill="#2D3240"
                        />
                      </svg>
                    </StyledComponent>
                  </styled.svg>
                </ArrowUp>
              </SortableArrow>
            </th>
          </StyledComponent>
        </Styled(styled.th)>
      </SortableHeadCell>
    </tr>
  </thead>
</table>
`;
